lib/msf/base/simple/exploit.rb

```diff
@@ -220,15 +220,12 @@ def self.job_check_proc(ctx)
     run_uuid = ctx[1]
     begin
       mod.setup
-      mod.framework.running << run_uuid
-      mod.framework.ready.delete run_uuid
+      mod.framework.job_state_tracker.start run_uuid
       result = mod.has_check? ? mod.check : Msf::Exploit::CheckCode::Unsupported
-      mod.framework.results[run_uuid] = {result: result}
+      mod.framework.job_state_tracker.completed(run_uuid, result)
     rescue => e
-      mod.framework.results[run_uuid] = {error: e.to_s}
+      mod.framework.job_state_tracker.failed(run_uuid, e)
       mod.handle_exception e
-    ensure
-      mod.framework.running.delete run_uuid
```

Comment:
Do we need this still? 👀

Reply:
it's handled in the job tracker now so don't think we should need it here

Reply:
please double check my logic on that though, but I feel like it's the kind of thing that should be abstracted into the tracker